src/extensions/renderer/base/coord-ele-math/coords.mjs

```diff
@@ -395,53 +395,81 @@ BRp.getAllInBox = function( x1, y1, x2, y2 ){
     if( ele.isNode() ){
       var node = ele;
       var eventsEnabled = node.pstyle('text-events').strValue === 'yes';
-      var boxSelectEnabled = node.pstyle('box-select-labels').strValue === 'yes';
-
-      var nodeBb = node.boundingBox({
-        includeNodes: true,
-        includeEdges: false,
-        includeLabels: boxSelectEnabled && eventsEnabled,
-      });
-
-      if (math.boundingBoxesIntersect(boxBb, nodeBb)) {
-        let rotatedLabelBox = getRotatedLabelBox(node);
-        let selectionBox = [
-          { x: boxBb.x1, y: boxBb.y1 },
-          { x: boxBb.x2, y: boxBb.y1 },
-          { x: boxBb.x2, y: boxBb.y2 },
-          { x: boxBb.x1, y: boxBb.y2 },
-        ];
-
-        if (math.satPolygonIntersection(rotatedLabelBox, selectionBox)) {
-          box.push(node);
+      var nodeBoxSelectMode = node.pstyle('box-selection').strValue;
+      
+      if( nodeBoxSelectMode !== 'none' ) {
+        var nodeBb = node.boundingBox({
+          includeNodes: true,
+          includeEdges: false,
+          includeLabels: nodeBoxSelectMode && eventsEnabled,
+        });
+
+        if (math.boundingBoxesIntersect(boxBb, nodeBb)) {
```

Comment:
This is the logic for box-selection:overlap for nodes

What about box-selection:contain for nodes?  Probably the bounds of the node body and rotated label box can just be checked for containment.  These functions would be useful for this: https://github.com/cytoscape/cytoscape.js/blob/unstable/src/math.mjs#L363-L372